src/software/simulated_tests/simulated_test_fixture.py

```diff
@@ -94,133 +107,157 @@ def time_provider(self):
-                        ValidationProtoSet, always_validation_proto_set
-                    )
+        self.__stopper()
 
-                # Check that all always validations are always valid
-                validation.check_validation(always_validation_proto_set)
 
-            # Check that all eventually validations are eventually valid
-            validation.check_validation(eventually_validation_proto_set)
+class InvariantTestRunner(SimulatedTestRunner):
 
-            __stopper()
+    """
+    Runs a simulated test only once with a given parameter
 
-        def excepthook(args):
-            """This function is _critical_ for show_thunderscope to work.
-            If the test Thread will raises an exception we won't be able to close
-            the window from the main thread.
+    Test passes or fails based on the outcome of this test
+    """
 
-            :param args: The args passed in from the hook
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
-            """
+    def run_test(
+        self,
+        setup=(lambda x: None),
+        params=[0],
+        inv_always_validation_sequence_set=[[]],
+        inv_eventually_validation_sequence_set=[[]],
+        test_timeout_s=3,
+        tick_duration_s=0.0166,  # Default to 60hz
+        **kwargs,
+    ):
+        """Run an invariant test
+
+        :param setup: Function that sets up the World state and the gamecontroller before running the test
+        :param params: List of parameters for each iteration of the test
+                        (this method only uses the first element)
+        :param inv_always_validation_sequence_set: Validation functions for invariant testing
+                                that should hold on every tick
+        :param inv_eventually_validation_sequence_set: Validation functions for invariant testing
+                                that should eventually be true, before the test ends
+        :param test_timeout_s: The timeout for the test, if any eventually_validations
+                                remain after the timeout, the test fails.
+        :param tick_duration_s: The simulation step duration
+
+        """
 
-            __stopper(delay=PAUSE_AFTER_FAIL_DELAY_S)
-            self.last_exception = args.exc_value
-            raise self.last_exception
+        threading.excepthook = self.excepthook
 
-        threading.excepthook = excepthook
+        setup(params[0])
 
         # If thunderscope is enabled, run the test in a thread and show
         # thunderscope on this thread. The excepthook is setup to catch
         # any test failures and propagate them to the main thread
         if self.thunderscope:
 
-            run_sim_thread = threading.Thread(target=__runner, daemon=True)
+            run_sim_thread = threading.Thread(target=self.runner, daemon=True)
             run_sim_thread.start()
             self.thunderscope.show()
             run_sim_thread.join()
```

Comment:
in thunderscope mode, it looks like we ignore the actual test timeouts and timeout in 3 seconds (default)

Reply:
yeah for some reason no args were passed before, added those